hrms/payroll/doctype/gratuity/gratuity.py

```diff
@@ -244,14 +244,14 @@ def get_gratuity_amount(self, experience: float) -> float:
 						* total_component_amount
 						* slab.fraction_of_applicable_earnings
 					)
-					years_left -= slab.to_year - slab.from_year
 					slab_found = True
 
 				elif self._is_experience_within_slab(slab, experience):
 					gratuity_amount += (
 						years_left * total_component_amount * slab.fraction_of_applicable_earnings
 					)
 					slab_found = True
+				years_left -= slab.to_year - slab.from_year
```

Comment:
The `years_left` calculation does not seem correct. In the case of `_is_experience_within_slab`, you should directly set `years_left = 0`.